pulp_container/app/registry_api.py

```diff
@@ -345,36 +349,132 @@ def get(self, request):
         return Response(data={})
 
 
-class CatalogView(ContainerRegistryApiMixin, APIView):
+class ContainerCatalogSerializer(ModelSerializer):
+    """
+    Serializer for Distributions in the _catalog endpoint of the registry.
+    """
+
+    class Meta:
+        model = models.ContainerDistribution
+        fields = ["base_path"]
+
+
+class ContainerCatalogPagination(BasePagination):
+    """
+    Pagination class to paginate repositories by names according to the registry api specification.
+    """
+
+    def paginate_queryset(self, queryset, request, view=None):
+        """
+        Analyse the pagination parameters and prepare the queryset.
+        """
+        try:
+            self.n = int(request.query_params.get("n"))
+        except Exception:
+            self.n = self.page_size
+        last = request.query_params.get("last")
+        self.url = request.build_absolute_uri()
+
+        if last:
```

Comment:
to be honest i don't have a 100% understanding on how `last` works, how is it handled the case when some non-existent vale has been provided in the request query param?

Reply:
Well i suspect, that is why you specify last and not "first of next query". It does not matter if it got deleted in the meantime. You just list, what comes later in alphabethical order.